go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -65,11 +67,15 @@ type NodeController struct {
 //  1. Setting up a VXLAN gateway and hooking to the OVN gateway
 //  2. Setting back annotations about its VTEP and gateway MAC address to its own object
 func newNodeController(
-	_ kube.Interface,
+	kube kube.Interface,
```

Comment:
This godoc text also needs to be changed

```
// newNodeController returns a new node controller that runs on linux.
// On linux there's two controller implementations. One runs from 
// ovnkube-node binary and the other from HO node binary and they
// prepare the OVN and SDN nodes for the HO tunnel respectively. 
```

The current godoc text fits better with `NodeController` above.

Also I think that only this method `newNodeController` should remain in this file.

The remaining of the code could probably be moved to a new `ovn_node_linux.go`

And then node `node_windows.go` should be `ho_node_windows.go` and `honode_linux.go` be `ho_node_linux.go`

This might be going overboard but I am just trying to structure things in a way that make sense.

Reply:
fixed